Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1141,6 +1141,8 @@ void FrameLoader::provisionalLoadStarted()
         if (RefPtr page = frame->page())
             page->didStartProvisionalLoad();
     }
+
+    m_errorOccurredInLoading = false;
```

Comment:
This seems like it's in the wrong place, and it needs a test that verifies a successful load after an unsuccessful load indicates no more load failure.

Reply:
As I mentioned in the other comment - this is for the provisional loading case and I have another place I set it in the committed load case.

How would you recommend writing such a test? How should I get the same frame to load a failing URL and then a valid URL?

Reply:
Oh - I misread this line when saying it's in the wrong place. Where do you think it should be when a brand new load starts @achristensen07?

Reply:
After looking more closely, I would guess `FrameLoader::load` is a better place.

Reply:
Although there are lots of those flavors - I could put it in all of them.